torch_xla/distributed/spmd/xla_sharding.py

```diff
@@ -640,3 +640,23 @@ def backward(ctx, grad_output):
 
 def xla_patched_nn_linear_forward(m, input):
   return XLAPatchedLinear.apply(input, m.weight, m.bias)
+
+
+def apply_backward_optimization_barrier(m: torch.nn.Module):
```

Comment:
Does this need to be applied to each layer of the model? What's the expected entrypoint for e.g. Llama2?

Reply:
Oh I missed your link to the transformers repo - thanks!